mission_control/src/mission.cpp

```diff
@@ -62,8 +63,12 @@ std::unique_ptr<Mission> Mission::FromMissionDefinition(const std::string& missi
 
 void Mission::stop() { tree_.haltTree(); }
 
-BT::NodeStatus Mission::getStatus() { return tree_.rootNode()->status(); }
+BT::NodeStatus Mission::getStatus() { return behaviorStatus_; }
 
 std::string Mission::getCurrentMissionDescription() { return description_; }
 
-BT::NodeStatus Mission::Continue() { return tree_.tickRoot(); }
+BT::NodeStatus Mission::Continue()
+{
+  behaviorStatus_ = tree_.tickRoot();
+  return behaviorStatus_;
```

Comment:
@patapufate meta, note for myself: this seems like a good place to map behavior tree state to mission state.